crypto/storage/client.go

```diff
@@ -0,0 +1,146 @@
+package storage
+
+import (
+	"context"
+	"crypto"
+	"fmt"
+	"github.com/nuts-foundation/nuts-node/core"
+	"github.com/nuts-foundation/nuts-node/crypto/storage/httpclient"
+	"github.com/nuts-foundation/nuts-node/crypto/util"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+const StorageAPIConfigKey = "external-store"
+const httpClientTimeout = 100 * time.Millisecond
+
+// APIClient implements the Storage interface. It uses a simple HTTP protocol to connect to an external storage server.
+// This server can either be a secret store itself, or proxy the request to a key store such as Hashicorp Vault or Azure Key Vault.
+// It allows us to keep the codebase clean and allow other parties to write their own adaptor.
+type APIClient struct {
+	httpClient *httpclient.ClientWithResponses
+}
+
+func (c APIClient) Name() string {
+	return "Crypto"
+}
+
+func (c APIClient) CheckHealth() map[string]core.Health {
```

Comment:
Since `CheckHealth()` performs calls to external resources, it might be nice to have it pass a context which canb e used for time-outs and cancellation.

Reply:
it implements a Healthcheckable interface. I consider it outside the scope of this PR to fix that. Perhaps open an issue for it?